server/container/util/src/main/java/org/apache/james/util/Size.java

```diff
@@ -49,6 +50,8 @@
     Long value;
 
     private Size(Unit unit, Long value) {
+        Preconditions.checkArgument(value > 0, "Maxsize must be positive");
```

Comment:
A size can be equal to zero

Reply:
By the test, it can be negative also.
I just move validate into `of` method which for our useage in UploadRoutes

Reply:
`Preconditions.checkArgument(value >= 0, "Maxsize must be positive");`